yt_dlp/extractor/digitalconcerthall.py

```diff
@@ -0,0 +1,201 @@
+                resp = self._parse_json(e.cause.read().decode(), None, fatal=False)
+                if resp:
+                    error = resp.get('extra_info') or resp.get('error_description') or resp.get('error')
+                    if error:
+                        msg += ': ' + error
+            raise ExtractorError('Unable to obtain token: ' + msg)
+        self.debug_out("token_response: " + json.dumps(token_response))
+        self._ACCCESS_TOKEN = token_response.get('access_token')
+        # now login
+        data = {
+            'grant_type': 'password',
+            'username': username,
+            'password': password,
+        }
+        try:
+            self._download_json(
+                self._OAUTH_URL,
+                None, 'Logging in', data=urlencode_postdata(data), headers={
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                    'Referer': self._LOGIN_URL,
+                    'Authorization': 'Bearer ' + self._ACCCESS_TOKEN
+                })
+        except ExtractorError as e:
+            msg = 'Unable to login'
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                resp = self._parse_json(e.cause.read().decode(), None, fatal=False)
+                if resp:
+                    error = resp.get('extra_info') or resp.get('error_description') or resp.get('error')
+                    if error:
+                        msg += ': ' + error
+            raise ExtractorError('Unable to log in: ' + msg)
+
+    def _real_initialize(self):
+        self._login()
+
+    def _real_extract(self, url):
+        MAX_TITLE_LENGTH = 128
+        language, video_id = re.match(self._VALID_URL, url).groups()
+        if not language:
+            language = 'en'
+        self.debug_out("url: " + url + " video_id: " + video_id + " language: " + language)
+        webpage = self._download_webpage(url, video_id)
+        playlist_title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title') or \
+            self._og_search_title(webpage)
+        self.debug_out("playlist_title: " + playlist_title)
+
+        # use the API to get other information about the concert
+        vid_info_dict = self._download_json(
+            'https://api.digitalconcerthall.com/v2/concert/'
+            + video_id, video_id, headers={'Accept': 'application/json',
+                                           'Accept-Language': language})
+        embedded = vid_info_dict.get('_embedded')
+        entries = []
+        for embed_type in embedded:
+            # embed_type should be either 'work' or 'interview'
+            # 'work' will be an array of one or more works
+            for item in embedded.get(embed_type):
+                if embed_type == 'interview':
+                    item['is_interview'] = 1
+                else:
+                    item['is_interview'] = 0
+                stream_href = item.get('_links').get('streams').get('href')
```

Comment:
Is this supposed to be fatal or not? If it is, just use []. Else use try_get/traverse_obj